lib/OpenQA/Schema/Result/Workers.pm

```diff
@@ -70,10 +68,10 @@ sub name {
     return $self->host . ":" . $self->instance;
 }
 
-sub seen {
-    my ($self, $workercaps, $error) = @_;
-    $self->update({t_seen => now()});
-    $self->update_caps($workercaps) if $workercaps;
+sub seen ($self, $options = {}) {
+    my $data = {t_seen => now()};
+    $data->{error} = $options->{error} if exists $options->{error};
+    $self->update($data);
```

Comment:
Supposedly it would be the easiest to just let the caller specify any data that should be updated and only set `t_seen` in addition:

```suggestion
sub seen ($self, $data = {}) {
    $data->{t_seen} = now();
    $self->update($data);
```

This way this function becomes never bigger, even if we want to set more additional data in the future.

Reply:
Do we need an additional abstraction here at all? I personally would try to keep it simple.

Reply:
Yes, `$worker->seen` is called in 5 places, copypasting `$worker->update({t_seen => now()})` for each one would be silly.

Reply:
But seriously, we've got performance issues to resolve, this is not the time to nitpick on tiny details like this. The only question you should be asking is if this is better than before.

Reply:
> Do we need an additional abstraction here at all?

With that question I didn't want to ask whether we should drop that function. You said:

> I don't think that would be good abstraction.

And I personally don't see `seen` as an additional layer of abstraction besides avoiding to repeat the code it contains. So I'm only questioning whether the statement "I don't think that would be good abstraction." is relevant.

> But seriously, we've got performance issues to resolve, this is not the time to nitpick on tiny details like this.

That's why I have never requested changes.

Reply:
I see, then we had a bit of miscommunication there. My comment about abstraction was a bit too vague indeed.

What i was referring to was specifically replacing a generic options hash `$options = {}` with a `$data = {}` hash tied to the fields of the table. Because it would severely limit how we can expand the method in the future. Instead of being able to add any arbitrary option to the method without changing its signature, we would limit the ways in which we can expand it for no real gain at all.